src/world/entity.rs

```diff
@@ -383,20 +406,38 @@ impl Generation {
     /// Panics in debug mode if it's not alive.
     fn die(&mut self) {
         debug_assert!(self.is_alive());
-        self.0 = -self.0;
+        self.0 = NonZeroU32::new(-self.id() as u32).map(Generation);
     }
 
     /// Revives and increments a dead `Generation`.
     ///
     /// # Panics
     ///
-    /// Panics in debug mode if it is alive.
-    fn raised(self) -> Generation {
-        debug_assert!(!self.is_alive());
-        Generation(1 - self.0)
+    /// Panics if it is alive.
+    fn raised(&self) -> Generation {
+        assert!(!self.is_alive());
+        Generation(unsafe { NonZeroU32::new_unchecked((1 - self.id()) as u32) })
```

Comment:
Note that wrapping will give the wrong behaviour here, sending id from `i32::MIN + 1` to `i32::MIN` and vice versa. I think it would be better to just unconditionally panic here. Potentially we could start wrapping the generations back around to 1 but that sounds risky to me.